lib/SIL/Parser/SILParsedLocalValueMap.h

```diff
@@ -0,0 +1,72 @@
+//===--- SILParser.h - SIL Undef Value Representation -----------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_SIL_PARSER_H
+#define SWIFT_SIL_PARSER_H
+
+#include "swift/AST/DeclContext.h"
+#include "swift/Basic/LLVM.h"
+#include "swift/Parse/Parser.h"
+#include "swift/SIL/SILInstruction.h"
+
+namespace swift {
+namespace syntax {
+
+//===----------------------------------------------------------------------===//
+// SILParser local value container
+//===----------------------------------------------------------------------===//
+
+struct UnresolvedValueName {
+  StringRef Name;
+  SourceLoc NameLoc;
+
+  bool isUndef() const { return Name == "undef"; }
+};
+
+/// Holds SIL local values and forward references.
+class SILParsedLocalValueMap {
+  SILModule &module;
+
+  llvm::StringMap<SILValue> localValues;
+  /// Used to keep track of forward referenced sil operands. These may be used
+  /// in basic blocks that aren't dominance ordered.
+  llvm::MapVector<StringRef, SourceLoc> forwardReferences;
+
+  /// Convenience method for diagnosing errors. Uses the module's ASTContext
+  /// diagnosis engine.
+  template <class... DiagArgs, class... Args>
+  InFlightDiagnostic diagnose(SourceLoc loc, Diag<DiagArgs...> diagID,
+                              Args &&... args);
+
+public:
+  llvm::function_ref<void()> hadError = []() {
```

Comment:
function_ref members are deeply suspicious.  This should be a std::function or const (or both if you can swing it).